test/e2e/scrapeconfig_test.go

```diff
@@ -2119,3 +2122,148 @@ var LightSailSDTestCases = []scrapeCRDTestCase{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           "VPS",
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Invalid ApplicationKey with empty value",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           "VPS",
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Invalid Service type",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           "InvalidService",
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Empty Endpoint",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           "VPS",
+					Endpoint:          ptr.To(""),
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Valid with RefreshInterval for supported version",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           "VPS",
+					RefreshInterval:   ptr.To(monitoringv1.Duration("30s")),
+				},
+			},
+		},
+		expectedError: false,
+	},
+	// {
+	// 	// Following should technically show error when the refreshInterval has invalid value for the supported version
+	// 	name: "Invalid RefreshInterval for unsupported version",
+	// 	scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+	// 		OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+	// 			{
+	// 				ApplicationKey:    "valid-app-key",
+	// 				ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+	// 				ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+	// 				Service:           "VPS",
+	// 				RefreshInterval:   ptr.To(monitoringv1.Duration("30s")),
+	// 			},
+	// 		},
+	// 	},
+	// 	expectedError: true,
+	// },
+	// {
+	// 	name: "Missing ConsumerKey",
```

Comment:
Not sure why this fails. Could you re-check?

Reply:
I rewrote the testcases again and the following two failed once more :-

```go
	{
		name: "Invalid ConsumerKey with empty value",
		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
				{
					ApplicationKey:    "valid-app-key",
					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
					ConsumerKey:       v1.SecretKeySelector{Key: ""},
					Service:           monitoringv1alpha1.OVHServiceVPS,
				},
			},
		},
		expectedError: true,
	},
	{
		name: "Missing ConsumerKey",
		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
				{
					ApplicationKey:    "valid-app-key",
					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
					Service:           monitoringv1alpha1.OVHServiceVPS,
				},
			},
		},
		expectedError: true,
	},
```
Are the consumerKey config validations in place ?

Reply:
There aren't any explicit validations at the application level. The CRD spec is expected to be generated such that the given field is required. For the CRD generation we make use of `kubebuilder`.  This might have to do with how `kubebuilder` handles `+structType:=atomic`. I am unsure about the cause of this behavior though. CC: @simonpasquier

Reply:
For now, we may have to skip this test.

Reply:
I see, I'll remove it for now then.